packages/angular/cli/src/utilities/package-tree.ts

```diff
@@ -44,20 +44,25 @@ export interface PackageTreeNode {
 
 export async function readPackageJson(packageJsonPath: string): Promise<PackageJson | undefined> {
   try {
-    return JSON.parse((await fs.promises.readFile(packageJsonPath)).toString());
+    return JSON.parse((await readFile(packageJsonPath)).toString());
   } catch {
     return undefined;
   }
 }
 
 export function findPackageJson(workspaceDir: string, packageName: string): string | undefined {
+  // See: https://github.com/angular/angular-cli/pull/18610#issuecomment-681980185
+  const originalRequireCache = require.cache;
+
   try {
-    // avoid require.resolve here, see: https://github.com/angular/angular-cli/pull/18610#issuecomment-681980185
-    const packageJsonPath = resolve.sync(`${packageName}/package.json`, { basedir: workspaceDir });
+    require.cache = {};
+    const workspaceRequire = createRequire(`${workspaceDir}/`);
 
-    return packageJsonPath;
+    return workspaceRequire.resolve(`${packageName}/package.json`);
   } catch {
     return undefined;
+  } finally {
+    require.cache = originalRequireCache;
   }
```

Comment:
Something similar to the following should work for resolve (`Module` imported from `node:module`):
```
  assert(
    '_pathCache' in Module && Module._pathCache && typeof Module._pathCache === 'object',
    'Node.js Module does not contain _pathCache',
  );

  // Avoid resolve cache problems, see: https://github.com/angular/angular-cli/pull/18610#issuecomment-681980185
  const originalPathCache = Module._pathCache;
  try {
    Module._pathCache = {};
    const workspaceRequire = createRequire(`${workspaceDir}/`);
    
    return workspaceRequire.resolve(`${packageName}/package.json`);
  } catch {
    return undefined;
  } finally {
    Module._pathCache = originalPathCache;
  }
```

Reply:
Looked a bit more and it looks like it will not solve https://github.com/angular/angular-cli/pull/18610#issuecomment-681980185